Source/Transforms/LMCHilbertTransform.cc

```diff
@@ -169,32 +212,22 @@ namespace locust
 
     fftw_complex* HilbertTransform::Transform(std::deque<double> FieldBuffer)
     {
-        int windowsize=FieldBuffer.size();
-
-        fftw_complex *originaldata;
-        originaldata = (fftw_complex *) fftw_malloc(sizeof(fftw_complex) * windowsize);
-        fftw_complex *SignalComplex;
-        SignalComplex = (fftw_complex*)fftw_malloc( sizeof(fftw_complex) * windowsize );
-        fftw_complex *FFTComplex;
-        FFTComplex = (fftw_complex*)fftw_malloc( sizeof(fftw_complex) * windowsize );
-        fftw_complex *hilbert;
-        hilbert = (fftw_complex *) fftw_malloc(sizeof(fftw_complex) * windowsize);
+        int windowsize=FieldBuffer.size()-1;
```

Comment:
Shouldn't the windowsize be the same as FieldBuffer.size()?  What does the `-1`  do here?

Reply:
It turns out that FieldBuffer.size() returns 1 more than the actual buffer size. So for a buffer size of 50, it returns 51. There is probably an off-by-one error upstream, but I'm not sure where it is.

Reply:
Actually, I think I know why. The buffer is initialized to have 50 zeros. Then it goes through LMCArraySignalGenerator::DriveAntenna() and FillBuffers is called. Then it has 51 entries. Then GetFIRSample is called, so all that is being done on a buffer that is 51 elements long. Only after that is PopBuffers called.
So either we have to initialize to 49, or pop the buffers before getting the FIR sample.

Reply:
Aha, I see.